File: RocketFitJavaOfficial_BackEnd/src/main/java/org/taboralemu/RocketFitJavaOfficial/DTOs/ExerciseRecordDTO.java
package org.taboralemu.RocketFitJavaOfficial.DTOs;

public class ExerciseRecordDTO {

	private String exercise_name;
	private int sets;
	private int reps;
	private double weight;
	private int auth_id;
	private int day;
	private int workout_number;

	public ExerciseRecordDTO() {

	}

	public ExerciseRecordDTO(String exercise_name, int sets, int reps, double weight, int auth_id, int day,
			int workout_number) {
		setExercise_name(exercise_name);
		setSets(sets);
		setReps(reps);
		setWeight(weight);
		setAuthId(auth_id);
		setDay(day);
		setWorkoutNumber(workout_number);
	}

	public String getExercise_name() {
		return exercise_name;
	}

	public int getSets() {
		return sets;
	}

	public int getReps() {
		return reps;
	}

	public double getWeight() {
		return weight;
	}

	public int getAuthId() {
		return auth_id;
	}

	public int getDay() {
		return day;
	}

	public int getWorkoutNumber() {
		return workout_number;
	}

	public void setExercise_name(String exercise_name) {
		this.exercise_name = exercise_name;
	}

	public void setSets(int sets) {
		this.sets = sets;
	}

	public void setReps(int reps) {
		this.reps = reps;
	}

	public void setWeight(double weight) {
		this.weight = weight;
	}

	public void setAuthId(int id) {
		this.auth_id = id;
	}

	public void setDay(int day) {
		this.day = day;
	}

	public void setWorkoutNumber(int workoutNumber) {
		this.workout_number = workoutNumber;
	}

}

File: RocketFitBackend_Python/RocketFitDjango2/RFApp/dtos/ExerciseRecordDTO.py


class ExerciseRecordDTO:
    """
    Class: Data Transfer Object class for Exercise Record objects.
    """

    exerciseName : str
    sets : int
    reps : int
    weight : int
    authId : int
    day : int
    workoutNumber : int
    targetWeight : int

    def __init__(self, **kwargs) -> None:
        self.exerciseName = kwargs["exerciseName"]
        self.sets = kwargs["sets"]
        self.reps = kwargs["reps"]
        self.weight = kwargs["weight"]
        self.authId = kwargs["authId"]
        self.day = kwargs["day"]
        self.workoutNumber = kwargs["workoutNumber"]
        self.targetWeight = kwargs["targetWeight"]
        self.createdDate = kwargs["createdDate"]

    def asdict(self):
        return {
            "exerciseName" : self.exerciseName,
            "sets" : self.sets,
            "reps" : self.reps,
            "weight" : self.weight,
            "authId" : self.authId,
            "day" : self.day,
            "workoutNumber" : self.workoutNumber,
            "targetWeight" : self.targetWeight,
            "createdDate" : self.createdDate
        }